src/thunderbots/software/grsim_communication/grsim_command_primitive_visitor.cpp

```diff
@@ -124,7 +146,34 @@ void GrsimCommandPrimitiveVisitor::visit(
 
 void GrsimCommandPrimitiveVisitor::visit(const KickPrimitive &kick_primitive)
 {
-    // TODO: https://github.com/UBC-Thunderbots/Software/issues/93
+    Point kick_origin    = kick_primitive.getKickOrigin();
+    Angle kick_direction = kick_primitive.getKickDirection();
+
+    double final_speed_at_destination = 0.0;
+    Point perp_line_to_shot           = Point::createFromAngle(kick_direction.quarter());
+    Point normal_line_to_shot         = Point::createFromAngle(kick_direction.half());
+    double dest_behind_x              = -cos(kick_direction.toDegrees());
+    double dest_behind_y              = -sin(kick_direction.toDegrees());
+    Point dest_behind = kick_origin + Vector(dest_behind_x, dest_behind_y);
+
+    // If current distance between robot and line perpendicular to direction of shot is
+    // less than that of between destination behind shot and the same perpendicular line
+    // while robot is in same direction as shot, can go for the shot
+    if (!(abs(offsetToLine(kick_origin, perp_line_to_shot, robot.position())) <=
+              abs(offsetToLine(kick_origin, perp_line_to_shot,
+                               Point(dest_behind_x, dest_behind_y))) &&
```

Comment:
I think you just want to use `dest_behind` here rather than `Point(dest_behind_x, dest_behind_y)`. They are not equivalent, because `dest_behind` is relative to the kick_origin.